src/Illuminate/Bus/DatabaseBatchRepository.php

```diff
@@ -239,19 +239,16 @@ public function delete(string $batchId)
      */
     public function prune(DateTimeInterface $before)
     {
-        $query = $this->connection->table($this->table)
+        $items = $this->connection->table($this->table)
             ->whereNotNull('finished_at')
-            ->where('finished_at', '<', $before->getTimestamp());
+            ->where('finished_at', '<', $before->getTimestamp())
+            ->cursor();
 
-        $totalDeleted = 0;
-
-        do {
-            $deleted = $query->take(1000)->delete();
-
-            $totalDeleted += $deleted;
-        } while ($deleted !== 0);
+        foreach ($items as $item) {
+            $item->delete();
```

Comment:
Memory issue should be solved, but will this not fire a query each time?

Reply:
I agree this will be much slower and I am also not convinced there is any memory issues with the previous version of the code which was taken from the official Paracelsus telescope package.

https://github.com/laravel/telescope/blob/master/src/Console/PruneCommand.php

Not sure if the original code was misread but is not loading any models into memory as it is just doing a delete query with a limit of 1000 records per delete.